File: src/main/csharp/p00/Problem0009.cs
namespace ProjectEuler.main.csharp.p00
{
    public class Problem0009
    {
        public static int Solve()
        {
            for (var x = 998; x > 0; x--)
            {
                var limit = (1000 - x) / 2;
                for (var y = 1000 - x - 1; y > limit; y--)
                {
                    var z = 1000 - x - y;
                    if (y * y + z * z == x * x) return x * y * z;
                }
            }

            return 1;
        }
    }
}

File: src/main/kotlin/p00/Problem0009.kt
package p00

class Problem0009 {

    fun solve(): Int {
        for (a in 2..998) {
            for (b in a - 1 downTo 1) {
                val c = 1000 - a - b
                if (a * a + b * b == c * c) return a * b * c
            }
        }

        return 0
    }

}